axis.go

```diff
@@ -355,11 +361,14 @@ type DefaultTicks struct{}
 var _ Ticker = DefaultTicks{}
 
 // Ticks returns Ticks in a specified range
-func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
+func (DefaultTicks) Ticks(min, max float64, format func(v float64, prec int) string) (ticks []Tick) {
 	const SuggestedTicks = 3
 	if max < min {
 		panic("illegal range")
 	}
+	if format == nil {
```

Comment:
instead of sprinkling `DefaultTicks` and `LogTicks` with these `if format == nil` checks, perhaps we could instead put this default inside `makeAxis` ?
Like we do for `Axis.Tick.Marker` ?

and expose `formatFloatTick` as `DefaultTickFormatter` ?

Reply:
Do you mean:

1. Creating an interface:

````
type TickFormatter interface {
    Format func(v float64, prec int) string
}
````

And having `Axis.Format` be of type `TickFormatter` and have a `DefaultTickFormat` struct implementing the interface and assigning that in `makeAxis`?

or

2. Simply create a `DefaultTickFormat` function that is exposed and gets assigned to `Axis.Format` (of type `func(v float64, prec int) string`) in `makeAxis`?

Reply:
(apologies for not being very clear)

the latter. this makes this new field (and its handling in the code) much more inline with the others.